ur_simulation_ignition/launch/ur_sim_control.launch.py

```diff
@@ -149,14 +149,6 @@ def launch_setup(context, *args, **kwargs):
         arguments=["joint_state_broadcaster", "--controller-manager", "/controller_manager"],
     )
 
-    # Delay rviz start after `joint_state_broadcaster`
-    delay_rviz_after_joint_state_broadcaster_spawner = RegisterEventHandler(
-        event_handler=OnProcessExit(
-            target_action=joint_state_broadcaster_spawner,
-            on_exit=[rviz_node],
-        )
-    )
```

Comment:
Can we please not remove this but add a condition instead `condition=IfCondition(launch_rviz),`?

Reply:
Sure, this does the trick. I've updated the launch file accordingly.